src/test/regress/expected/drop_partitioned_table.out

```diff
@@ -0,0 +1,469 @@
+--
+-- DROP_PARTITIONED_TABLE
```

Comment:
I think these tests are great, but we don't seem to have any explicit test that shows we skip sending `DROP TABLE partition_name;`, which is our ultimate goal. Now, we assume that it'll happen when the `normal` and `original` values are in the expected state.

Can you  maybe add a failure test that intercepts the `DROP TABLE partition_name;` and after this if `DROP TABLE parent` happens, it is not triggered?

Or, can you think of any other way of testing it?

Reply:
> Now, we assume that it'll happen when the normal and original values are in the expected state.
Well, isn't that enough? We check the `normal` and `original` values through the trigger in the test, meaning we check that we called `citus_drop_all_shards` with the correct value of `drop_shards_metadata_only` parameter.

Reply:
What if we introduce a bug in `citus_drop_all_shards ` -- or there is already a bug? We'd never realize. I think we should definitely add a test showing we are skipping to send DROP partitions, otherwise it is likely that one day we'll break this.

Reply:
If there is already a bug in `citus_drop_all_shards`, it is out of scope of this PR to fix that. Essentially, I am only adding a parameter to skip some lines of code that send remote commands: checking that I pass the right value of this parameter to the function seems enough testing to me.

Maybe if we introduce a bug in the future? Let me think about this and also think of a way to write a test about this.

By the way, we don't skip sending `DROP TABLE partition_name;` we skip sending remote commands to drop shards, i.e. remote `DROP TABLE partition_name_shardid`.

Reply:
Sorry but I cannot follow the reason for the pushback here. These tests do not explicitly show what this PR is all about: Skip sending `DROP TABLE partition_name_shardid`

What if this PR has a bug on calculating `applyRemoteShardsDrop`? We have no tests that ensures it. Am I missing anything here?

Reply:
After Teams discussion, I am adding a test that explicitly shows we don't send remote commands. In this way we ensure the behavior now and for possible future bugs or code refactoring. Thanks Önder.